sound/soc/sof/ipc4.c

```diff
@@ -183,18 +185,19 @@ int sof_ipc4_tx_message(struct snd_sof_ipc *ipc, u32 header, u32 extension,
 }
 EXPORT_SYMBOL(sof_ipc4_tx_message);
 
-void snd_sof_ipc4_msgs_rx(struct snd_sof_dev *sdev, u32 msg, u32 msg_ext)
+void snd_sof_ipc4_msgs_rx(struct snd_sof_dev *sdev)
 {
+	struct sof_ipc4_msg *ipc4_msg = sdev->ipc->msg.reply_data;
 	int err;
 
-	if (!SOF_IPC4_GLB_NOTIFY_MSG_TYPE(msg))
+	if (!SOF_IPC4_GLB_NOTIFY_MSG_TYPE(ipc4_msg->primary))
 		return;
 
-	switch (SOF_IPC4_GLB_NOTIFY_TYPE(msg)) {
+	switch (SOF_IPC4_GLB_NOTIFY_TYPE(ipc4_msg->primary)) {
 	case SOF_IPC4_GLB_NOTIFY_FW_READY:
 		/* check for FW boot completion */
 		if (sdev->fw_state == SOF_FW_BOOT_IN_PROGRESS) {
-			err = sof_ops(sdev)->fw_ready(sdev, msg);
+			err = sof_ops(sdev)->fw_ready(sdev, ipc4_msg->primary);
```

Comment:
Can you help to add debug log for default case to notify unsupported reply ? thanks!

Reply:
Right, should I add a new patch on top or a new PR? We can only print the primary and extension for now, but we should have similar logging as with IPC3.

Reply:
@RanderWang, something like this?
https://github.com/ujfalusi/sof-linux/commit/41ebbc9c34242cb19432765be137f564038671c0

Reply:
sure

Reply:
you can create a new PR if you like